### DISCUSSION TITLE
Mike and legacy (non-mkdocs) website

### DISCUSSION THREAD

**Discussion by jponge** [squidfunk/mkdocs-material#4097] [Q&A] [answered]
Hi, I'm in the process of migrating a project documentation to MkDocs Material and use Mike.

I've read both projects documentation but it's not clear to me how to _switch_ but keep a version alias for the previous website that we had (plain Jekyll).
I have other projects where we started with MkDocs/Mike from scratch, but it's not the case here.

Going forward I'd like to have Mike manage versions, but I'd like to setup Mike so that I can have a version for the previous Jekyll-rendered content which currently lives in the `gh-pages` branch.

Thanks for any pointer you may have 🙏

**Comment by squidfunk** [comment, marked as answer]
Excellent question and definitely a sound use case. Material for MkDocs only consumes what `mike` generates, i.e. the `versions.json`. Theoretically, this file could include versions or links to other websites, so the user would be taken to that site when the respective version is selected. However, I'm not sure whether `mike` supports this, so maybe ask on their issue tracker.

As a quick idea, you could theoretically post-process the `versions.json`, add the links manually and then just add a redirect for that version to an external site. This could all be done with some scripting and post-processing, but as said, I find that to be a sound use case so definitely something that could be discussed with the maintainers of `mike`. If `mike` would support that, I'd be happy to make any changes necessary to get that working with Material for MkDocs.

**Reply by jponge** [reply]
Thanks for these notes!

Meanwhile I've tested deploying to:

- a clean branch: `mike` does add all files as needed
- an existing `gh-pages` branch: `mike` adds files (including `versions.json`) but does not overwrite any existing `index.html`, which may be an issue.

Perhaps that's asking too much of `mike` TBH, it's probably sound to eiher
- clean the existing `gh-pages` branch, or
- manually move the existing files in a folder, then as you said edit `versions.json` afterwards.

**Reply by squidfunk** [reply]
I guess it's currently not possible without some manular work, patching in what you want after `mike` has been deployed. As I suggested, maybe something to discuss with the maintainers of `mike`, since I believe that this might come up more often.

**Reply by jponge** [reply]
See https://github.com/jimporter/mike/issues/110
